internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1478,6 +1478,20 @@ func (v *VRGInstance) reconcileAsPrimary() {
 func (v *VRGInstance) pvcsDeselectedUnprotect() error {
 	log := v.log.WithName("PvcsDeselectedUnprotect")
 
+	if !(v.instance.Spec.ReplicationState == ramendrv1alpha1.Primary &&
+		!v.instance.Spec.PrepareForFinalSync &&
+		!v.instance.Spec.RunFinalSync) {
+		log.Info(
+			"PVC deselection skipped",
+			"replicationstate",
+			v.instance.Spec.ReplicationState,
+			"finalsync",
+			v.instance.Spec.PrepareForFinalSync || v.instance.Spec.RunFinalSync,
+		)
+
+		return nil
+	}
+
```

Comment:
For this and the next addition we can use common function, something like this:

```
func shouldSkipPVCAction(spec ramendrv1alpha1.VolRepGroupSpec, log logr.Logger, action string) bool {
	if spec.ReplicationState != ramendrv1alpha1.Primary ||
		spec.PrepareForFinalSync ||
		spec.RunFinalSync {
		log.Info(
			fmt.Sprintf("PVC %s skipped", action),
			"replicationstate", spec.ReplicationState,
			"finalsync", spec.PrepareForFinalSync || spec.RunFinalSync,
		)
		return true
	}
	return false
}
```

Reply:
Tracked [here](https://github.com/RamenDR/ramen/issues/2098)